app/src/main/java/com/sunasterisk/getitdone/ui/detail/DetailFragment.kt

```diff
@@ -0,0 +1,260 @@
+import com.sunasterisk.getitdone.data.repository.TaskRepository
+import com.sunasterisk.getitdone.data.source.local.TaskListLocalDataSource
+import com.sunasterisk.getitdone.data.source.local.TaskLocalDataSource
+import com.sunasterisk.getitdone.data.source.local.dao.TaskDAOImpl
+import com.sunasterisk.getitdone.data.source.local.dao.TaskListDAOImpl
+import com.sunasterisk.getitdone.data.source.local.database.AppDatabase
+import com.sunasterisk.getitdone.utils.Constants.BUNDLE_TASK
+import com.sunasterisk.getitdone.utils.Constants.REQUEST_KEY_DELETE_TASK
+import com.sunasterisk.getitdone.utils.Constants.REQUEST_KEY_UPDATE_TASK
+import com.sunasterisk.getitdone.utils.Constants.STATUS_COMPLETED
+import com.sunasterisk.getitdone.utils.Constants.STATUS_NOT_COMPLETE
+import com.sunasterisk.getitdone.utils.formatToString
+import com.sunasterisk.getitdone.utils.showDateTimePicker
+import com.sunasterisk.getitdone.utils.toDate
+import com.sunasterisk.getitdone.utils.toast
+import kotlinx.android.synthetic.main.fragment_detail.*
+import java.util.*
+
+class DetailFragment : BaseFragment<DetailContract.View, DetailPresenter>(),
+    DetailContract.View, View.OnClickListener {
+
+    override val layoutId get() = R.layout.fragment_detail
+
+    private var taskId = -1
+
+    private var taskLists = mutableListOf<TaskList>()
+    private var task: Task? = null
+
+    private var presenter: DetailPresenter? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let { taskId = it.getInt(ARG_TASK_ID) }
+    }
+
+    override fun initComponents(savedInstanceState: Bundle?) {
+        context?.let {
+            val database = AppDatabase.getInstance(it)
+
+            val taskListRepository = TaskListRepository.getInstance(
+                TaskListLocalDataSource.getInstance(
+                    TaskListDAOImpl.getInstance(database)
+                )
+            )
+
+            val taskRepository = TaskRepository.getInstance(
+                TaskLocalDataSource.getInstance(
+                    TaskDAOImpl.getInstance(database)
+                )
+            )
+
+            presenter =
+                DetailPresenter(taskListRepository, taskRepository).apply {
+                    attach(this@DetailFragment)
+                    getTaskFromId(taskId)
+                    getAllTaskList()
+                }
+        }
+
+        btnBack.setOnClickListener(this)
+        btnChangeStatus.setOnClickListener(this)
+        btnDelete.setOnClickListener(this)
+
+        imageImportant.setOnClickListener(this)
+
+        myDayLayout.setOnClickListener(this)
+        datePickerLayout.setOnClickListener(this)
+
+        editTextDescription.doOnTextChanged { text, _, _, _ ->
+            text?.let { string ->
+                if (string.isBlank()) imageDescription.setColorFilter(Color.DKGRAY)
+                else task?.let { imageDescription.setColorFilter(Color.parseColor(it.color)) }
```

Comment:
Nếu không viết được trên 1 dòng thì phải thêm đẩy đủ cặp {} vào trong câu lệnh 
```suggestion
                if (string.isBlank()) {
                    imageDescription.setColorFilter(Color.DKGRAY)
                } else {
                    task?.let { 
                        imageDescription.setColorFilter(Color.parseColor(it.color)) 
                    }
```

Reply:
```suggestion
                val color = if (string.isBlank()) Color.DKGRAY else task?.color 
                color?.let(imageDescription::setColorFilter)
```